elastic/eui
# [Beta] Add `EuiCollapsibleNavItem` component

## Summary

This PR sets up a new `EuiCollapsibleNavItem` component that follows the new nav redesign and UX outlined in https://github.com/elastic/eui/issues/6759.

This PR, as noted by the [Beta] label in the title, is not a complete feature in and of itself. However, I'm choosing to ship it to main for the following reasonns reasons:

- Creating new components/folders does not interfere or cause interruptions to consumers using the old components, and is relatively risk free
- Releasing it (unannounced/undocumented) allows @tsullivan to incrementally use redesign progress in his Kibana feature work and battle-test our component using production data, leading to issues (hopefully) getting caught faster

## QA

- `gh pr checkout 6904`
- `yarn storybook` (may need to `yarn` beforehand if you haven't run EUI locally in a while)
- Go to http://localhost:6006/?path=/story/euicollapsiblenavbeta--kibana-example
- Go to http://localhost:6006/?path=/story/euicollapsiblenavitem--edge-case-testing
- Click around / QA the various links and displays and confirm they look and behave coherently (in particular, sub-sub-item nesting)

### General checklist

~- [ ] Checked in **mobile**~ Responsive/mobile work isn't yet done, will be part of the next upcoming work where the `EuiCollapsibleNavBeta` component itself gets fleshed out more

Changelog & docs: Skipping this for now due to beta status & until we have something we're ready for all public consumers to use. Once all components are put together, we can deprecate the old component and update docs. 
~- [ ] A **[changelog](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/documenting/changelogs.md)** entry exists and is marked appropriately~
~- [ ] Props have proper **autodocs** (using `@default` if default values are missing) and **[playground toggles](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/documenting/playgrounds.md)**~
~- [ ] Added **[documentation](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/documenting)**~

Figma: See linked issue with Figma URLs - will likely need to update EUI's Figma library after this work is all over
~- [ ] Updated the **[Figma](https://www.figma.com/community/file/964536385682658129)** library counterpart~
